lib/describe.js

```diff
@@ -81,11 +81,56 @@ class Describer {
         }
     }
 
+    _describeTime(time) {
+        if (time.isAbsolute) {
+            const date = new Date;
+            date.setHours(time.hour);
+            date.setMinutes(time.minute);
+            date.setSeconds(time.second);
+            if (time.second !== 0) {
+                return this._format.timeToString(date, {
+                    hour: 'numeric',
+                    minute: '2-digit',
+                    second: '2-digit'
+                });
+            } else {
+                return this._format.timeToString(date, {
+                    hour: 'numeric',
+                    minute: '2-digit'
+                });
+            }
+        } else {
+            switch (time.relativeTag) {
+                case 'morning':
+                    return this._('the morning');
+                case 'evening':
+                    return this._('the evening');
+                default:
+                    return time.relativeTag;
+            }
+        }
+    }
+
     _describeDate(date, operator, offset) {
         let base;
 
         if (date === null) {
             base = this._("now");
+        } else if (date.isDatePiece) {
+            let year = this._(date.year === null ?
+                              "????" :
+                              date.year.toString().padStart(4, '0'));
+            let month = this._(date.month === null ?
+                               "??" :
+                               date.month.toString().padStart(2, '0'));
+            let day = this._(date.day === null ?
+                             "??" :
+                             date.day.toString().padStart(2, '0'));
+            let time = this._(date.time === null ?
+                              "??:??" :
+                              this._describeTime(date.time));
+            base = this._interp(this._("${year}-${month}-${day} ${time}"),
+                                { year, month, day, time });
```

Comment:
This description is not readable, so it's not suitable. "describe.js" is about deterministic human-readable representation of the code, used to confirm the agent's behavior.